src/EnergyPlus/api/datatransfer.cc

```diff
@@ -316,60 +392,120 @@ int getPluginTrendVariableHandle(const char* name) {
+        EnergyPlus::ShowContinueError("The getPluginTrendVariableValue function will return 0 to allow the plugin to finish, then EnergyPlus will abort");
+        EnergyPlus::PluginManagement::shouldIssueFatalAfterPluginCompletes = true;
+        return 0;
     }
     return EnergyPlus::PluginManagement::pluginManager->getTrendVariableValue(handle, timeIndex);
 }
 
 Real64 getPluginTrendVariableAverage(int handle, int count) {
     if (handle < 0 || handle > EnergyPlus::PluginManagement::pluginManager->maxTrendVariableIndex) {
-        throw std::runtime_error("Plugin Trend Variable index out of range");
+        // need to fatal out once the plugin is done
+        // throw an error, set the fatal flag, and then return
+        EnergyPlus::ShowSevereError("Data Exchange API: Problem -- index error in getPluginTrendVariableAverage; received handle: " + std::to_string(handle));
+        EnergyPlus::ShowContinueError("The getPluginTrendVariableAverage function will return 0 to allow the plugin to finish, then EnergyPlus will abort");
+        EnergyPlus::PluginManagement::shouldIssueFatalAfterPluginCompletes = true;
+        return 0;
     }
     if (count < 2 || count > ((int)EnergyPlus::PluginManagement::pluginManager->getTrendVariableHistorySize(handle) - 1)) {
-        throw std::runtime_error("Plugin Trend Variable history count out of range");
+        // need to fatal out once the plugin is done
+        // throw an error, set the fatal flag, and then return
+        EnergyPlus::ShowSevereError("Data Exchange API: Problem -- trend history count argument out of range in getPluginTrendVariableAverage; received value: " + std::to_string(count));
+        EnergyPlus::ShowContinueError("The getPluginTrendVariableAverage function will return 0 to allow the plugin to finish, then EnergyPlus will abort");
+        EnergyPlus::PluginManagement::shouldIssueFatalAfterPluginCompletes = true;
+        return 0;
     }
     return EnergyPlus::PluginManagement::pluginManager->getTrendVariableAverage(handle, count);
 }
 
 Real64 getPluginTrendVariableMin(int handle, int count) {
     if (handle < 0 || handle > EnergyPlus::PluginManagement::pluginManager->maxTrendVariableIndex) {
-        throw std::runtime_error("Plugin Trend Variable index out of range");
+        // need to fatal out once the plugin is done
+        // throw an error, set the fatal flag, and then return
+        EnergyPlus::ShowSevereError("Data Exchange API: Problem -- index error in getPluginTrendVariableMin; received handle: " + std::to_string(handle));
+        EnergyPlus::ShowContinueError("The getPluginTrendVariableMin function will return 0 to allow the plugin to finish, then EnergyPlus will abort");
+        EnergyPlus::PluginManagement::shouldIssueFatalAfterPluginCompletes = true;
+        return 0;
     }
     if (count < 2 || count > ((int)EnergyPlus::PluginManagement::pluginManager->getTrendVariableHistorySize(handle) - 1)) {
```

Comment:
Same here, what if the min value is in timeIndex ==1, use count < 1.

Reply:
Oh, I see, the array size should be >1 but you'll still get the min value of all elements.

Reply:
I hear you on the trend index, I'll allow a lookup of just the first value.

Reply:
Yeah, on the trend functions that look over a list, I think they should require at least two.  Getting the min/max/average/direction/sum of a single value seems dumb.  But looking up the value at the current time spot seems good.  Is it OK just just modify the "getValue" function to allow 1 and require the others to be >= 2?

Reply:
Yes, I agree.

Reply:
I assume that anytime these trend arrays are allocated the size is > 1. I hope that's always true.

Reply:
Huh.  I think I see a problem.  I appreciate your comments on this.

This shouldn't be comparing `count > size - 1`, this should just be comparing `count > size`.

Imagine if I create a trend variable with 3 history terms.  Then I call to get the min over each time count:
 - With count == 1, this will abort, because why would you call `min` for just one value.
 - With count == 2, this will calculate the min over the first and second values
 - With count == 3, this will abort, because it would have compared `count > size - 1`, which is `3 > 2`, and thought it was out of range.

I'll go through and clean these up!